cp1616/src/cp1616_io_controller.cpp

```diff
@@ -48,17 +53,29 @@ Cp1616IOController::Cp1616IOController() :
   input_module_count_(0),
   output_module_count_(0),
   input_module_total_data_size_(0),
-  output_module_total_data_size_(0)
+  output_module_total_data_size_(0),
+  num_of_input_modules_(0),
+  num_of_output_modules_(0)
 {
-  //Allocate memory for Input module variables
-  input_module_data_length_.resize(NUM_OF_INPUT_MODULES);
-  input_module_state_.resize(NUM_OF_INPUT_MODULES);
-  input_module_address_.resize(NUM_OF_INPUT_MODULES);
-
-  //Allocate memory for Output module variables
-  output_module_data_length_.resize(NUM_OF_OUTPUT_MODULES);  
-  output_module_state_.resize(NUM_OF_OUTPUT_MODULES);
-  output_module_address_.resize(NUM_OF_OUTPUT_MODULES);
+  //Parse data from yaml config file
+  std::string filepath;
+  nh->getParam("filepath", filepath);
```

Comment:
Instead of creating a class that requires the ROS node handle, you should create a ROS agnostic class that is constructed with the information you need.  In this case, the file path should be in the constructor argument.  The node handle and parameter reading can then exist in your `main` or node class.